docs/architecture.rst

```diff
@@ -1,66 +1,167 @@
 
-TODO
+Standard color representations as described in :ref:`terminology` are predefined in :class:`.TQECColor`.
 
-``tqec.templates``
-------------------
+Some conventional implementations of computations in ``.dae`` format are available in the :ref:`gallery-reference-label`.
 
-TODO
+:mod:`.computation`
+--------------------
+
+Defines data structures for the high-level :class:`.BlockGraph` representations of a fault-tolerant computation protected by the surface code.
+
+The ``3D`` structures discussed in detail in :ref:`terminology` are defined in this module.
 
-``tqec.plaquette``
-------------------
+* A :class:`.CorrelationSurface` represents a set of parity measurements between the input and output logical operators.
+* A :class:`.Cube` is a fundamental building block constituting of a block of quantum operations that occupy a specific spacetime volume.
+  Quantum information encoded in the logical qubits can be preserved or manipulated by these blocks.
+* A :class:`.Pipe` is a block that connects :class:`.Cube` objects in a :class:`.BlockGraph` but does not occupy spacetime volume on its own.
+* :class:`.PipeKind` helps determine the kind of a pipe in a  :class:`.BlockGraph` based on the wall bases at the head of the pipe in
+  addition to a Hadamard transition.
+* :class:`.Port` depicts the open ports in a :class:`.BlockGraph`.
+* A :class:`.YHalfCube` represents Y-basis initialization and measurements.
+* :class:`.ZXCube` defines cubes with only X or Z basis boundaries.
 
-TODO
 
-``tqec.circuit``
+:mod:`.compile`
 ----------------
 
-TODO
+Responsible for translations between internal representations.
+
+:class:`.BlockGraph` from the functionality in :ref:`interop_ref` is further translated into a compiled graph through :func:`.compile_block_graph`.
+
+.. mermaid::
+    :align: center
+    :config: {"theme": "base", "darkMode": "true"}
+
+    %%{
+    init: {
+        'theme': 'base',
+        'themeVariables': {
+        'primaryColor': '#AFEEEE',
+        'lineColor': '#6495ED'
+        }
+    }
+    }%%
+
+    graph
+        A[BlockGraph] --> B[compiled graph] --> C[ScheduledCircuit]--> D[stim.Circuit]
+
+Multiple block builder protocols defined in ``tqec/compile/observables/builder.py`` will take the high-level structure of a block to
+templates and plaquettes i.e. a fully annotated ``stim.Circuit``.
+
+:mod:`.templates`
+^^^^^^^^^^^^^^^^^
+
+Generates an array of numbers in a representation of a 2-dimensional, scalable, arrangement of plaquettes. This allows us to describe a
+circuit that can scale the desired code distance.
+
+
+:mod:`.plaquette`
+^^^^^^^^^^^^^^^^^
+
+A plaquette is the representation of a local quantum circuit. A surface code patch implements one layer in time or one round of the surface code.
+
+This module maps from the numbered templates to some plaquettes that implement small local circuits to measure a stabilizer as a :class:`.ScheduledCircuit` instance.
```

Comment:
Not sure this is needed/accurate.

Reply:
The second sentence was supposed to be in the next section. Thanks for pointing this out.